src/stream.ts

```diff
@@ -160,11 +155,37 @@ export class WebRTCStream implements Stream {
       if (closed || this.writeClosed) {
         break;
       }
-      let res = buf.subarray();
-      let send_buf = pb.Message.toBinary({ message: buf.subarray() });
-      log.trace(`[stream:${this.id}][${this.stat.direction}] sending message: length: ${res.length} ${res}, encoded through pb as ${send_buf}`);
-      this.channel.send(send_buf);
+      const res = buf.subarray();
+      const msgbuf = pb.Message.toBinary({message: buf.subarray()});
+      const sendbuf = lp.encode.single(msgbuf)
+      log.trace(`[stream:${this.id}][${this.stat.direction}] sending message: length: ${res.length} ${res}, encoded through pb as ${msgbuf}`);
+      this.channel.send(sendbuf.subarray())
```

Comment:
More of a question, but why do you have to call `subarray` all the time? Is it to switch type somehow?

Reply:
The `Uint8ArrayList` can have a single message spread across multiple different `Uint8Array` instances. Calling `subarray` makes them readable as a single `Uint8Array` which is much easier to work with.